### ISSUE TITLE
[HOLD for payment 2023-05-04] ON HOLD [$1000] Suggestion text below company address is not visible when we don't select any suggested option but it is visible in every other cases

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#16448]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Open the app
2. Open Workspaces
3. Open any workspace
4. Open connect bank account
5. Click on connect manually
6. Fill in step 1 information (eg data: routing number = 210021002, acc no = 1234)
7. In company address, type in address but don't select any option and after typing, select city below
8. Observe that suggestion text below company address is gone
9. Fill in other details, click on continue and click on back and obserce that suggestion text below company address is back
10. Click on back few times and click on startover, again fill in step 6 details and observe that company address has suggestion text below it
11. Type in address and select any one suggestion and observe that company address has suggestion text below it

## Expected Result:
App should display suggestion text below company address if user decides to fill it on their own and not use the suggested addresses

## Actual Result:
App does not display suggestion text below company address if user decides to fill it on their own and if suggestions were still displayed before user tries to move to fill in other details

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [x] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.2.88-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/227250842-bc253546-6f19-4bd5-97b1-2778e8171123.mp4

https://user-images.githubusercontent.com/43996225/227250929-6267a11f-4548-4ddf-be78-3398e62e9764.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @dhanashree-sawant
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1679566976110789

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~018ed613b675853bc5</li>
        <li>Upwork Job ID: 1641411262190702592</li>
        <li>Last Price Increase: 2023-03-30</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @zanyrenney (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by zanyrenney** [comment]
I'm not sure if this constitutes a bug, so need to check on this and bring a conversation to bug0.

**Comment by zanyrenney** [comment]
I guess looking at this again, it is inconsistent so makes sense to progress forward here.

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~018ed613b675853bc5

**Comment by MelvinBot** [comment]
Current assignee @zanyrenney is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @eVoloshchak (`External`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @iwiznia (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by zanyrenney** [comment]
Thanks for your proposal @Prince-Mendiratta

**Comment by Pujan92** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Suggestion text not toggles and shows once we move focus to the other elements without selecting/pressing any suggested option

### What is the root cause of that problem?
Currently, onPress of the option from the `GooglePlacesAutocomplete` we are setting the `displayListViewBorder` to false which shows the hint text. But on blur of the textInput of `GooglePlacesAutocomplete` textInputProps we are not setting `displayListViewBorder` to false which causes this issue.

### What changes do you think we should make in order to solve the problem?
We need to update the onBlur inside `textInputProps` of the `GooglePlacesAutocomplete` with 
```onBlur: () => setDisplayListViewBorder(false),```
https://github.com/Expensify/App/blob/b091e3c747dc39e9642c65199ee31d46b09d81ef/src/components/AddressSearch.js#L213-L236
We are not passing onBlur from the parent for any occurrences so we can take out `props.onBlur`.

**Comment by tienifr** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
App does not display hint text under the company address is the user click outside of the dropdown without selecting an option.

### What is the root cause of that problem?
In this line https://github.com/Expensify/App/blob/4ee21e95a38858df28e46caf54dc97e1bca9bc7a/src/components/AddressSearch.js#L236, the hint is only displayed if the `displayListViewBorder` is false.

The `displayListViewBorder` is set to false if the user selects an option, but is not if the user blurs out of the `GooglePlacesAutocomplete`.

### What changes do you think we should make in order to solve the problem?

We need to set the `displayListViewBorder` to false inside `InteractionManager.runAfterInteractions` when the user blurs the address search input and the option list.

This can be done by updating this line https://github.com/Expensify/App/blob/4ee21e95a38858df28e46caf54dc97e1bca9bc7a/src/components/AddressSearch.js#L241 to 
```
onBlur: (event) => {
    if (containerRef.current && event.relatedTarget && !containerRef.current.contains(event.relatedTarget)) { 
        setDisplayListViewBorder(false);
    }
    props.onBlur(event);
}
```

The `containerRef` is the ref of the address search container (could be [this one](https://github.com/Expensify/App/blob/4ee21e95a38858df28e46caf54dc97e1bca9bc7a/src/components/AddressSearch.js#L201)) that we can set via `useRef`.

The related target check is required here because without it, when we select an option, the `onBlur` will still trigger, setting `displayListViewBorder` to false will make the auto complete component re-render before `onPress` is called, making `selecting an option` not working.

Currently there's a bug in our `react-native-google-places-autocomplete` fork here https://github.com/Expensify/react-native-google-places-autocomplete/blob/e12768f1542e7982d90f6449798f0d6b7f18f192/GooglePlacesAutocomplete.js#L860 that is not passing the `event` to `onBlur`, we need to fix that as well. The main `react-native-google-places-autocomplete` lib already fixed is so we can just update our fork to the latest

### What alternative solutions did you explore? (Optional)
- We can wrap the `setDisplayListViewBorder(false);` inside `InteractionManager.runAfterInteractions` as well instead of using related target check
- The `relatedTarget` check can be added upstream in the library instead of in our code

### Result
Working well after the fix:

https://user-images.githubusercontent.com/113963320/229275934-9e6971a6-6c5e-4688-9716-afb4be087812.mov

**Comment by MelvinBot** [comment]
@iwiznia, @eVoloshchak, @zanyrenney Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by iwiznia** [comment]
@eVoloshchak can you review the proposals above please?

**Comment by eVoloshchak** [comment]
@Prince-Mendiratta's and @Pujan92's proposals use the same approach: call `setDisplayListViewBorder(false)` when GooglePlacesAutocomplete's input is blurred. They both resolve our issue, but as @tienifr's pointed out in their proposal, there is a bug: _when we select an option, the onBlur will still trigger, setting displayListViewBorder to false will make the auto-complete component re-render before onPress is called, making selecting an option not working._ So essentially you can't select an address

@tienifr's [proposal](https://github.com/Expensify/App/issues/16448#issuecomment-1492881279) addresses this issue.

> We need to set the displayListViewBorder to false inside InteractionManager.runAfterInteractions when the user blurs the address search input and the option list.

Not sure I fully understand, do you propose we use both `InteractionManager.runAfterInteractions` and `relatedTarget` check or just one of these?

**Comment by tienifr** [comment]
@eVoloshchak sorry for the confusion, either one of these will fix the issue, but I prefer to use the relatedTarget check

**Comment by iwiznia** [comment]
Proposal looks good

**Comment by tienifr** [comment]
Hi @eVoloshchak @iwiznia We need to update react-native-google-places-autocomplete in our fork before. [The PR on react-native-google-places-autocomplete fork](https://github.com/Expensify/react-native-google-places-autocomplete/pull/3) is ready for review

**Comment by tienifr** [comment]
@iwiznia Friendly bump, could you take a look at 2 PRs on this issue? Thanks

**Comment by iwiznia** [comment]
Reviewed the fork and posted in slack to see how we handle the unsigned commits

**Comment by zanyrenney** [comment]
Sounds good. Putting on hold now.

**Comment by MelvinBot** [comment]
@iwiznia, @eVoloshchak, @zanyrenney, @tienifr Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by eVoloshchak** [comment]
Not overdue
We should remove the signed commits requirement in all of the forks soon, which would allow us to merge the PR for this issue

**Comment by eVoloshchak** [comment]
This shouldn't be on HOLD anymore, the first PR was merged, only https://github.com/Expensify/App/pull/16875 is left, currently in review

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.3.6-0 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/16875


If no regressions arise, payment will be issued on 2023-05-04. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@eVoloshchak] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@eVoloshchak] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@eVoloshchak] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@eVoloshchak] Determine if we should create a regression test for this bug.
- [ ] [@eVoloshchak] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@zanyrenney] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by dhanashree-sawant** [comment]
Hi @zanyrenney , @iwiznia , @eVoloshchak ping for payment.

**Comment by zanyrenney** [comment]
I was OOO. Catching up now

**Comment by zanyrenney** [comment]
Currently having an issue on Upwork with the Job posting: 
![2023-05-09_14-01-28](https://github.com/Expensify/App/assets/56830058/c2a5f960-f1f8-4732-90bc-f8e59a163fec)

**Comment by eVoloshchak** [comment]
- The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/9207
- The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: https://github.com/Expensify/App/pull/9207#issuecomment-1540489201
- A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: I don't think additional discussion is needed, this is a visual bug, there is nothing we can do besides automated UI tests
-  Determine if we should create a regression test for this bug.
_Is it easy to test for this bug? **Yes**
Is the bug related to an important user flow? (For example, adding a bank account) **Yes**
Is it an impactful bug? **No**_

    The bug has a very low impact, but it's easy to test and is related to `connect bank account` flow. I think we might want to add a simple regression test

**Comment by eVoloshchak** [comment]
**Regression Test Proposal**

1. Open Workspaces > select any workspace > connect bank account
2. Click on connect manually
3. Fill in step 1 information (eg data: routing number = 210021002, acc no = 1234)
4. In the company address, type in address but don't select any option, wait for the list with addresses to load
5. Click outside the list (ex: click on the city field)
6. Verify that the suggestion text below the company address is displayed

**Do we agree 👍 or 👎**

**Comment by zanyrenney** [comment]
At Accountex so won't get to this regression test + payout for a few days, reassiging.

**Comment by conorpendergrast** [comment]
Cool I'll get to this today!

**Comment by conorpendergrast** [comment]
@sophiepintoraetz Sorry to bring you in on this too Sophie! I am OoO until Monday and in the interest of these payments being overdue, I wanted to make sure they didn't have to wait until then. 

Once you and @iwiznia have agreed if the bonuses are due, then the [contracts here](https://github.com/Expensify/App/issues/16448#issuecomment-1524681939) can be paid. I've already paid the issue reporter!

**Comment by iwiznia** [comment]
Yep, agree on bonuses

**Comment by conorpendergrast** [comment]
Ionatan is speedy, so I'm doing the payments. [Wrap-up steps ](https://github.com/Expensify/App/issues/16448#issuecomment-1524681965)are yours, Sophie!

**Comment by conorpendergrast** [comment]
Payments done

**Comment by sophiepintoraetz** [comment]
QA steps requested [here](https://github.com/Expensify/Expensify/issues/283059).
